synapseclient/models/wiki.py

```diff
@@ -1 +1,804 @@
+            wiki_id=self.id,
+            synapse_client=synapse_client,
+        )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Attachment_Handles: Owner ID {self.owner_id}, Wiki ID {self.id}, Wiki Version {self.wiki_version}"
+    )
+    async def get_attachment_handles_async(
+        self,
+        *,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> list:
+        """
+        Get the file handles of all attachments on this wiki page asynchronously.
+
+        Arguments:
+            synapse_client: Optionally provide a Synapse client.
+        Returns:
+            The list of FileHandles for all file attachments of this WikiPage.
+        Raises:
+            ValueError: If owner_id or id is not provided.
+        """
+        if not self.owner_id:
+            raise ValueError("Must provide owner_id to get attachment handles.")
+        if not self.id:
+            raise ValueError("Must provide id to get attachment handles.")
+
+        return await get_attachment_handles(
+            owner_id=self.owner_id,
+            wiki_id=self.id,
+            wiki_version=self.wiki_version,
+            synapse_client=synapse_client,
+        )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Attachment_URL: Owner ID {self.owner_id}, Wiki ID {self.id}, File Name {kwargs['file_name']}"
+    )
+    async def get_attachment_url_async(
+        self,
+        file_name: str,
+        *,
+        redirect: Optional[bool] = False,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> dict:
+        """
+        Get the URL of a wiki page attachment asynchronously.
+
+        Arguments:
+            file_name: The name of the file to get.
+            redirect: When set to false, the URL will be returned as text/plain instead of redirecting. Default is False.
+            synapse_client: Optionally provide a Synapse client.
+        Returns:
+            The URL that can be used to download a file for a given WikiPage file attachment.
+        Raises:
+            ValueError: If owner_id or id is not provided.
+        """
+        if not self.owner_id:
+            raise ValueError("Must provide owner_id to get attachment URL.")
+        if not self.id:
+            raise ValueError("Must provide id to get attachment URL.")
+        if not file_name:
+            raise ValueError("Must provide file_name to get attachment URL.")
+
+        return await get_attachment_url(
+            owner_id=self.owner_id,
+            wiki_id=self.id,
+            file_name=file_name,
+            wiki_version=self.wiki_version,
+            redirect=redirect,
+            synapse_client=synapse_client,
+        )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Attachment_Preview_URL: Owner ID {self.owner_id}, Wiki ID {self.id}, File Name {kwargs['file_name']}"
+    )
+    async def get_attachment_preview_url_async(
```

Comment:
Are `get_markdown_url_async`, `get_attachment_preview_url_async`, `get_attachment_url_async` methods that a user is intended to call?

It seems to me like the use-case here is that instead - The user wants to get the file by making these calls

Reply:
Yes, this is a good point. I will try to address this. Thanks